src/platform/backends/lxd/lxd_request.h

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2019 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef MULTIPASS_LXD_REQUEST_H
+#define MULTIPASS_LXD_REQUEST_H
+
+#include <QFile>
+#include <QEventLoop>
+#include <QBuffer>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QTimer>
+#include <QUrl>
+#include <QUrlQuery>
+#include <QtNetwork/QNetworkAccessManager>
+#include <QtNetwork/QNetworkRequest>
+#include <QtNetwork/QNetworkReply>
+#include <QtNetwork/QSslCertificate>
+#include <QtNetwork/QSslConfiguration>
+#include <QtNetwork/QSslKey>
+
+#include <multipass/optional.h>
+#include <multipass/format.h>
+#include <multipass/logging/log.h>
+
+#include <QDebug>
+
+namespace mp = multipass;
+namespace mpl = multipass::logging;
+
+namespace
+{
+typedef std::exception LXDNotFoundException;
+
+constexpr auto request_category = "lxd request";
+
+auto make_ssl_config()
+{
+    QSslConfiguration ssl_config;
+    ssl_config.setLocalCertificate(QSslCertificate::fromPath("/home/ubuntu/.config/lxc/client.crt")[0]);
+    QFile keyfile{"/home/ubuntu/.config/lxc/client.key"};
+    keyfile.open(QIODevice::ReadOnly);
+    ssl_config.setPrivateKey(QSslKey(keyfile.readAll(), QSsl::Rsa));
+    ssl_config.setPeerVerifyMode(QSslSocket::VerifyNone);
+
+    return ssl_config;
+}
```

Comment:
As discussed on IRC, we can set up a `QSslConfiguration` object in the `LXDVirtualMachineFactory` ctor and then pass that to each `LXDVirtualMachine` to avoid having to do the set up at every request.